### ISSUE TITLE
Leaf piles from Sakura saplings are very destructive

### ISSUE THREAD

**Issue by saltyseadoggo** [TerraformersMC/Terrestria#181]
**Expected Behaviour**
The leaf piles generated by Sakura saplings when they grow would only be able to replace things that the player can place blocks over, such as grass or ferns.
**Actual Behaviour**
The leaf piles generated by Sakura saplings when they grow are very destructive in what blocks that they replace. I tested to confirm that they are able to replace torches, flower pots, bamboo saplings, buttons, pressure plates, signs, banners, redstone dust and lanterns, but they are likely to be able to replace many more.
**Attachments**
![2020-10-03_15 23 43](https://user-images.githubusercontent.com/69029346/95002963-c398d480-058e-11eb-89cc-fd7eb7429e04.png)
![2020-10-03_15 23 58](https://user-images.githubusercontent.com/69029346/95002964-c4316b00-058e-11eb-9fcc-0d4fc3861a69.png)
![2020-10-03_15 27 36](https://user-images.githubusercontent.com/69029346/95002966-c4ca0180-058e-11eb-8e7e-b418b9adbaf3.png)
![2020-10-03_15 31 05](https://user-images.githubusercontent.com/69029346/95002967-c5629800-058e-11eb-8e58-782a79feb2b6.png)

**Comment by coderbot16** [comment]
Oh, wow, that definitely is an oversight. Thanks for the report! It looks like https://github.com/TerraformersMC/Terrestria/blob/1.16/src/main/java/com/terraformersmc/terrestria/feature/tree/treedecorators/SakuraTreeDecorator.java#L37-L41 is the culprit:

```java
//check for a solid block and place on top
if (world.getBlockState(mutable).isOpaque() || world.getFluidState(mutable).isIn(FluidTags.WATER)) {
	world.setBlockState(mutable.up(), TerrestriaBlocks.SAKURA_LEAF_PILE.getDefaultState(), 3);
	break;
}
```

Basically, it places a leaf pile on top of any opaque block, without checking if the above block can be replaced by trees (such as with `canTreeReplace`). That would explain the behavior that you're seeing.

**Comment by coderbot16** [comment]
Confirmed that this was fixed in the 1.16.3 Terrestria versions.
